src/julia_gc.c

```diff
@@ -646,7 +646,7 @@ static void GapTaskScanner(jl_task_t * task, int root_task)
         if (tag->bits.gc & 2)
             rescan = 0;
     }
-    if (stack && tid < 0) {
+    if (stack) {
```

Comment:
This seems like the right thing to do....

In fact `tid < 0` means `tid == -1` means that the task is not the current task of any thread. Why was it ever OK to skip scanning the stack of such tasks? It might have been the previously active task on a thread, right? And then skipping it could lead to us missing a GC root and hence a GC crash, no?

Or is it not that bad? Then what am I missing this time?

Reply:
You aren't missing anything, that was a bug.